tests/test_tree.py

```diff
@@ -0,0 +1,36 @@
+from typing import TypeAlias
```

Comment:
just a thought for the future: would be great to see simple unit tests as doctests for functions

Reply:
Maybe if we found well isolated functions for that.